### PR TITLE
improve `Pkg.add` output

### PR DESCRIPTION
**Pull request by wildart**
This PR improves `Pkg.add` output:
- no `PkgError` needed for information message
- do not catch LibGit2 exceptions

### CODE REVIEW

File: base/pkg/entry.jl [hunk 1]
```diff
@@ -50,19 +50,20 @@ function add(pkg::AbstractString, vers::VersionSet)
     outdated = :maybe
     @sync begin
         @async if !edit(Reqs.add,pkg,vers)
-            ispath(pkg) || throw(PkgError("unknown package $pkg"))
```
Comment (tkelman) [inline base/pkg/entry.jl]: I think this should stay an error

Comment (wildart) [inline base/pkg/entry.jl]: It's a top level function (`Pkg.add` just wrapper around it) that serves as Pkg UI. So why do we need to dump a stack trace just to inform user that package does not exist?

**Comment by wildart** [comment]
I am ok with throwing exceptions in internal Pkg API (`PkgError`, `GitError`), but we need to differentiate between different types of errors. I'm going to close this an redo whole thing with enhanced exception type.

File: base/pkg/entry.jl (hunk 1 above)
Comment (tkelman) [inline base/pkg/entry.jl]: Stack trace is useful if you want to dig into why Pkg works the way it does when the error gets thrown, even if it's something seemingly obvious like a package that doesn't exist. It also shouldn't silently succeed here.

Comment (wildart) [inline base/pkg/entry.jl]: I disagree. It is a bad practice to clutter user interface with unwanted information. It creates confusion and sense of instability.

Comment (tkelman) [inline base/pkg/entry.jl]: Tried to add a package, package didn't exist. That's an error.

Comment (wildart) [inline base/pkg/entry.jl]: No, it is not error. It's just happened that the required package is not registered with METADATA.

Comment (tkelman) [inline base/pkg/entry.jl]: That's an error. "Just happened" that we failed to do what the user asked for.

**Pull request closed without merging**